root/static/scripts/common/components/ExpandedArtistCredit.js

```diff
@@ -0,0 +1,68 @@
+/*
+ * @flow
+ * This file is part of MusicBrainz, the open internet music database.
+ * Copyright (C) 2019 MetaBrainz Foundation
+ * Licensed under the GPL version 2, or (at your option) any later version:
+ * http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import React from 'react';
+
+import commaOnlyList from '../i18n/commaOnlyList';
+import {reduceArtistCredit} from '../immutable-entities';
+
+import ArtistCreditLink from './ArtistCreditLink';
+import DescriptiveLink from './DescriptiveLink';
+
+export const ExpandedArtistCreditList = ({
+  artistCredit,
+}: {artistCredit: ArtistCreditT}) => {
+  if (!artistCredit) {
+    return null;
+  }
+
+  const names = artistCredit.names;
+  const artistList = [];
+  let showList = false;
+
+  if (reduceArtistCredit(artistCredit) !== names[0].artist.name ||
+    names.length > 1 || names[0].artist.comment) {
```

Comment:
I think I'd just replace this condition with
```JS
if (names.some(x => x.artist.name !== x.name || x.artist.comment)) {
```
and remove the whole `showList` thing. It looks like the code is using this condition to avoid running all the stuff below if not necessary, but has some false positives making the `showList` thing is necessary. But I'd think we could just encode the complete logic here - it's probably not any slower than running `reduceArtistCredit` + dealing with any false positives.

Reply:
Seems sensible, changed!